woocommerce/woocommerce-gutenberg-products-block
# Register a global local pickup method for the block based checkout

Targetting the https://github.com/woocommerce/woocommerce-blocks/pull/7383 branch since this is a follow-up.

This PR shows how we can register a new shipping method (in this case called _pickup_location_) globally, not within zones, with its own settings screen and support for multiple rates.

- Wordage is up for discussion.
- I've based the table of locations on the designs and shipping zones UI, but rather than take the "modal" interface from woo core, I've just implemented inline editing to avoid using Backbone JS.
- Styles and JS are inline for now. Could be refactored.
- Hides address in the totals area when using collection
- Reverts the change we added to pull local pickup rates out of zones.

### Screenshots

![Screenshot 2022-10-18 at 12 08 09](https://user-images.githubusercontent.com/90977/196413666-2b8033e1-9664-4411-8ca2-3366ede27f10.png)

When pickup is selected:

![Screenshot 2022-10-18 at 12 08 12](https://user-images.githubusercontent.com/90977/196413710-73376dc8-89cb-4ab1-8175-37ece8f4d77e.png)

### Testing

1. Go to Settings > Shipping > Pickup Locations
2. Enable the method and insert some locations
3. On block based checkout, confirm the locations you added are selectable as shipping rates
4. Select a rate. Confirm address is hidden in sidebar/totals


### WooCommerce Visibility
